R/aes.r

```diff
@@ -143,13 +143,37 @@ print.uneval <- function(x, ...) {
   new_aes(NextMethod())
 }
 
-# Rename American or old-style aesthetics name
-rename_aes <- function(x) {
-  # Convert prefixes to full names
-  full <- match(names(x), ggplot_global$all_aesthetics)
-  names(x)[!is.na(full)] <- ggplot_global$all_aesthetics[full[!is.na(full)]]
+#' Standardise aesthetic names
+#'
+#' This function standardises aesthetic names by converting `color` to `colour`
+#' (also in substrings, e.g. `point_color` to `point_colour`) and translating old style
+#' R names to ggplot names (eg. `pch` to `shape`, `cex` to `size`).
+#' @param x Character vector of aesthetics names, such as `c("colour", "size", "shape")`.
+#' @return Character vector of standardised names.
+#' @keywords internal
+#' @export
+standardise_aes_names <- function(x) {
+  # convert US to UK spelling of colour
+  x <- sub("color", "colour", x, fixed = TRUE)
 
-  plyr::rename(x, ggplot_global$base_to_ggplot, warn_missing = FALSE)
+  # convert old-style aesthetics names to ggplot version
+  plyr::revalue(x, ggplot_global$base_to_ggplot, warn_missing = FALSE)
+}
+
+# x is a list of aesthetic mappings, as generated by aes()
+rename_aes <- function(x) {
+  names(x) <- standardise_aes_names(names(x))
+  duplicated_names <- names(x)[duplicated(names(x))]
+  if (length(duplicated_names) > 0L) {
+    duplicated_names_message <- paste0(
+      "`", duplicated_names, "`", collapse = ", "
+    )
+    warning(
```

Comment:
Wasn't this previously the responsibility of `plyr::rename()`?  Shouldn't duplicated names also be checked in `standardise_aes_names()`? Should we also check for code like `aes(colour=1,colour=2,colour=3)`?

Reply:
Yes, this code was taken from `plyr::rename()`. Because I now have a function that converts names rather than a named list it didn't make sense to call `plyr::rename()` anymore. Also, this gives us the opportunity to customize the warning message to be more meaningful.

Whether duplicated names should be checked in `standardise_aes_names()` is not clear to me. I think an argument can be made either way. If the job of `standardise_aes_names()` is to simply take a vector of strings and standardize each of them then it doesn't necessarily have the job to check for duplicates. Maybe there's a downstream use case where somebody needs aesthetics names standardized without having duplicates checked. On the other hand, moving the check into that function would work just fine with the current code base.

Regarding the case of multiple identical aesthetics, it does get caught by the same check:
``` r
> aes(colour=1, colour=2, colour=3)
Aesthetic mapping: 
* `colour` -> 1
* `colour` -> 2
* `colour` -> 3
Warning message:
Standardisation of aesthetics has created duplicates for the following: `colour`, `colour` 
```
So maybe it's just a matter of making the error message a little clearer. I think printing out the complete list of final aesthetic names, rather than just the ones that are duplicated, might be helpful.